modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/search/CPDefinitionIndexer.java

```diff
@@ -837,6 +837,10 @@ else if (!cpInstances.isEmpty()) {
 			}
 
 			for (CPInstance cpInstance : cpInstances) {
+				if (cpInstance.isExpired()) {
+					continue;
+				}
```

Comment:
Did you check the indexing for Control Panel -> Products (All/Skus) view? The expired products/skus should still be shown in there.

Reply:
To confirm, once you deploy the change here, rerun the indexer, in Control Panel-> Search -> Index Actions -> Reindex Search Indexes.